api/app/signals/apps/signals/v1/serializers.py

```diff
@@ -360,6 +360,21 @@ def _update_notes(self, instance, validated_data):
 
             Signal.actions.create_note(note_data, instance)
 
+    def validate(self, attrs):
+        if 'category_assignment' in attrs:
+            user = self.context['request'].user
+
+            # A user who does not have the SIA_BACKOFFICE permission (and thus has limited access to
+            # categories) should not be able to move a signal to a category he doesn't have access
+            # to.
+            if not user.has_perm('signals.' + permissions.SIA_BACKOFFICE) and \
+                    not user.has_perm('signals.' + attrs['category_assignment'][
+                        'sub_category'].permission.codename):
```

Comment:
Same as before. What if the category is renamed or altered?

Reply:
Nothing, because this permission is linked with the category id, not with the name.